### ISSUE TITLE
Issue with all content: Missing link underlines

### ISSUE THREAD

**Issue by aardrian** [mdn/yari#2901]
MDN URL: https://developer.mozilla.org/en-US/docs/ (and all pages below)

#### Issue

None of the links within the content have an underline. They are difficult to distinguish from the surrounding content and require interaction to identify them in some cases.

The page content is dark gray (`#212121`) and the link color is dark blue (`#00458b`). with a contrast ratio of 1.7:1, it is fair to assume most users with good vision under ideal conditions will have to take a second look.

As it stands now, the content links would fail WCAG 2.1 [Success Criterion 1.4.1 Use of Color](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html). The non-normative [G182](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20160317/G182) recommends using an additional visual cue besides color. [G183](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20160317/G183.html) recommends a 3:1 contrast ratio and an additional visual cue on focus or hover, but is now mooted with the advent of touch interfaces. I go into more detail about alternatives to underlines in [_On Link Underlines_](https://adrianroselli.com/2016/06/on-link-underlines.html).

#### Recommended fix

To restore underlines to links in the content…

In the file:
/node_modules/@mdn/minimalist/sass/atoms/_typography.scss

At line 69, please strike `text-decoration: none;`. Reference code with - preceding the line to strike:

```diff
  /* text links */
  a:link,
  a:visited {
    color: $link-color;
-    text-decoration: none;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
    }
  }
```

In the file:
/node_modules/@mdn/minimalist/sass/atoms/_links.scss

At line 3, please strike `text-decoration: none;`. Reference code with - preceding the line to strike:

```diff
  a {
    color: $link-color;
-    text-decoration: none;

    &:hover,
    &:focus {
      text-decoration: underline;
    }
```

This will restore underlines to links in the navigation and elsewhere as well. If you want to remove those, I suggest more targeted removal of underlines scoped to just those areas to ensure underlines are retained on links outside the content.

If you would like to try the content with restored underlines without fiddling in the browser's dev tools, I have bookmarklet that will restore underlines (which I tested on MDN) at http://rosel.li/bookmarklets#underlines.

#### Specific section or headline?

Throughout.

#### What did you expect to see?

Underlines on links.

#### Did you test this? If so, how?

I tried to use the content.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/element`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/web/html/element/index.html
* Last commit: https://github.com/mdn/content/commit/145e9f6f3d388fd2277aaf47f0b68175e4f9dcb5
* Document last modified: 2020-12-18T22:31:45.000Z

</details>

**Comment by schalkneethling** [comment]
Thank you @aardrian for reporting this. It is a BIG oversight. Whoever picks this up, here is what you are going to need to change:

https://github.com/mdn/yari/blob/200521ca2a9fe9a6e0a77295e3d2d5a2d4e8b6bb/client/src/document/index.scss#L46

```
a {
    text-decoration: underline;

    &:active {
      background-color: $primary-50;
      color: $text-color-inverted;
    }
  }
```

**Comment by peterbe** [comment]
> To avoid duplicated efforts on this, I assigned this issue to @dhananjayd1729 in the meantime.

Please don't assign issues. Not even to people in the core team. We assign them to ourselves or perhaps in some exceptions, for each other after we've had offline conversations. 

Contributors are more than welcome to make pull requests but they have to take the chance that a PR might come in slightly later than another contributors PR. 

All too often we've seen it happen over the years that well-meaning contributors show interest in solving an issue (and sometimes get assigned) but the PR never comes. 

To us, assignment is similar to putting something on our "weekly" to-do list. It's not ownership or best-matched domain/area.

**Comment by mattiapontonio** [comment]
In my opinion, it could be an anarchist attempt in which everyone can self-assign issues and even steal them from others and see if the team manages itself as the Agile theorists theorize.

---


Aside from that, you should investigate where the links have been unsubscribed and not add places where they are underlined (although it may seem like a maintenance gesture). And it would be nice to give a touch of style to the underlining perhaps with mathematically reasoned values.

**Comment by dhananjayd1729** [comment]
I have been trying to run 'yari' on my pc but I'm getting an error. I followed the instructions given in the readme file. I'm attaching ss, please have a look. I would appreciate the help.
![githuberror](https://user-images.githubusercontent.com/61632269/107932947-2621c880-6fa4-11eb-9026-8b425b43800a.PNG)

**Comment by schalkneethling** [comment]
> I have been trying to run 'yari' on my pc but I'm getting an error. I followed the instructions given in the readme file. I'm attaching ss, please have a look. I would appreciate the help.

Did you clone the content repo at the URL below to that location @dhananjayd1729?
https://github.com/mdn/content

**Comment by dhananjayd1729** [comment]
Yes, I did. I cloned 'content' and 'yari' repo in the 'mozilladocs' folder.

**Comment by schalkneethling** [comment]
> Yes, I did. I cloned 'content' and 'yari' repo in the 'mozilladocs' folder.

Ah! Just noticed. You need to add `/files/` so, `/content/files/`

**Comment by Ryuno-Ki** [comment]
Also, is the `C:\c\` in the screenshot on purpose?
(I've got a similar case on Matrix at the moment)

**Comment by schalkneethling** [comment]
> Yes, I did. I cloned 'content' and 'yari' repo in the 'mozilladocs' folder.

All good now @dhananjayd1729?

**Comment by dhananjayd1729** [comment]
> > Yes, I did. I cloned 'content' and 'yari' repo in the 'mozilladocs' folder.
> 
> All good now @dhananjayd1729?

I will try in a bit and will comment here.

**Comment by dhananjayd1729** [comment]
@schalkneethling .
![newerror](https://user-images.githubusercontent.com/61632269/108042759-3dc18580-7066-11eb-8d2f-1db0bf10b374.PNG)

**Comment by schalkneethling** [comment]
@dhananjayd1729 That is interesting. I am not entirely sure on that one, I reckon either @fiji-flo or @peterbe will be able to assist here.

On a side note, are you planning on working on this issue? In other words, is that why you are getting all of this set-up? Or is it unrelated?

**Comment by dhananjayd1729** [comment]
Thanks, @schalkneethling. Yes, I'm planning to work on this issue and would like to work on more issues in the future.

**Comment by fiji-flo** [comment]
@dhananjayd1729 what's your `CONTENT_ROOT` when you're running this?

Can you run `yarn tool gather-git-history`?

**Comment by dhananjayd1729** [comment]
@fiji-flo CONTENT_ROOT=<path-to-clone-of-content >/files . I didn't run 'yarn tool gather-git-history' yet.

**Comment by fiji-flo** [comment]
There's a lack of context.

I assume you checked out https://github.com/mdn/yari to `%USERPROFILE%\mozilladocs\yari` and https://github.com/mdn/content to `%USERPROFILE%\mozilladocs\content`

In that case make sure your `CONTENT_ROOT` is set to `%USERPROFILE%\mozilladocs\content\files`.

Could you then try to run `yarn tool gather-git-history` from within `%USERPROFILE%\mozilladocs\yari`.

**Comment by dhananjayd1729** [comment]
Yeah, I checked both clone of 'content' and 'yari' to %USERPROFILE%\mozilladocs\content and %USERPROFILE%\mozilladocs\yari
respectively. Okay, I will run yarn tool gather-git-history from %USERPROFILE%\mozilladocs\yari.

**Comment by schalkneethling** [comment]
@dhananjayd1729 also ensure `CONTENT_ROOT` is set as follows:

```
CONTENT_ROOT=%USERPROFILE%\mozilladocs\content\files
```

**Comment by schalkneethling** [comment]
@dhananjayd1729 You up and running?

**Comment by dhananjayd1729** [comment]
> @dhananjayd1729 also ensure `CONTENT_ROOT` is set as follows:
> 
> ```
> CONTENT_ROOT=%USERPROFILE%\mozilladocs\content\files
> ```

What do you mean by %USERPROFILE% ?(is it my github id or C:Users/..)
Also instead of make it messy here can we chat on matrix?

**Comment by Ryuno-Ki** [comment]
On Windows, `%SOMETHING%` refers to an environment variable „SOMETHING”.
`%UserProfile%` (since Windows is case insensitive, you might have seen this notation) should point to `C:\Users\daund` in your case.

**Comment by dhananjayd1729** [comment]
> @dhananjayd1729 what's your `CONTENT_ROOT` when you're running this?
> 
> Can you run `yarn tool gather-git-history`?


After running this command I've got this error.
![yarngather1](https://user-images.githubusercontent.com/61632269/108101264-fdd1c100-70ac-11eb-8098-8f6bdd60aabb.PNG)

**Comment by peterbe** [comment]
@dhananjayd1729 Can you run that again but with this:
```
yarn tool gather-git-history --verbose
```

It won't solve the immediate problem but next time, it'll print out a log stacktrace that shows exactly in which file that happened.

**Comment by dhananjayd1729** [comment]
> @dhananjayd1729 You up and running?

No, I'm stuck even after 'content root' points to files folder in a clone of 'content' .

**Comment by peterbe** [comment]
> > @dhananjayd1729 You up and running?
> 
> No, I'm stuck even after 'content root' points to files folder in a clone of 'content' .

Another important fix landed today relating to Windows and that `gather-git-history`. Can you try with the latest `main` again?

**Comment by dhananjayd1729** [comment]
> > > @dhananjayd1729 You up and running?
> > 
> > 
> > No, I'm stuck even after 'content root' points to files folder in a clone of 'content' .
> 
> Another important fix landed today relating to Windows and that `gather-git-history`. Can you try with the latest `main` again?

'Latest main' as in?

**Comment by peterbe** [comment]
We used to have the default branch be called `master`. Now it's called `main`.

**Comment by dhananjayd1729** [comment]
So you are basically suggesting me to change the directory to content? Because yari is the master branch at this stage.

**Comment by peterbe** [comment]
> So you are basically suggesting me to change the directory to content? Because yari is the master branch at this stage.

I know it's confusing with all the branches and stuff. It takes some getting used to. 
But if it's really tricky, consider starting over a bit and seeing if a fresh clone helps.

**Comment by Ryuno-Ki** [comment]
@dhananjayd1729 In https://github.com/mdn/yari/discussions/2951 @peterbe explained how to update a local setup to use `main` on Yari, too.

**Comment by peterbe** [comment]
> Thank you @aardrian for reporting this. It is a BIG oversight. Whoever picks this up, here is what you are going to need to change:
> 
> https://github.com/mdn/yari/blob/200521ca2a9fe9a6e0a77295e3d2d5a2d4e8b6bb/client/src/document/index.scss#L46
> 
> ```
> a {
>     text-decoration: underline;
> 
>     &:active {
>       background-color: $primary-50;
>       color: $text-color-inverted;
>     }
>   }
> ```

I tried that but it only underlines the `a` within the articles. And the headline links too. 
Other features such as the document footer, sidebars, breadcrumbs, and home page are unaffected. 
<img width="1237" alt="Screen Shot 2021-03-02 at 9 04 22 AM" src="https://user-images.githubusercontent.com/26739/109660087-74070680-7b36-11eb-9441-a456354e68b7.png">

I'm too old-school to understand why minimalist even removes this. I think it should be reset/default and then we can make exceptions on top. Like: `h2 a, h3 a { text-decoration:none}`

**Comment by aardrian** [comment]
Since the content links are the ones that need the underlines, not affecting footer, sidebars, and breadcrumbs seems fine. I agree that the headings likely do not need underlines (though I do like the pattern of an icon to indicate they can be used as anchor links for sharing, making them distinct from content links that navigate someone).

**Comment by schalkneethling** [comment]
> Other features such as the document footer, sidebars, breadcrumbs, and home page are unaffected.

That is fine and the intent of the change here.

> I'm too old-school to understand why minimalist even removes this.

I did consider it but, I think we will have to undo more underlining than we currently will have to add exceptions where we _do_ want underlines.

**Comment by schalkneethling** [comment]
@dhananjayd1729 were you able to get up and running?

**Comment by dhananjayd1729** [comment]
@schalkneethling I was having exams back then so I couldn't look into it. But yeah from today onwards I will try to run again.

**Comment by schalkneethling** [comment]
> @schalkneethling I was having exams back then so I couldn't look into it. But yeah from today onwards I will try to run again.

Ok, let me know if you run into problems or have any questions.

**Comment by aardrian** [comment]
Glad to see this wrapped up, thanks all!

How long before it appears on the site? I just purged my cache to be sure it is not already out there before asking.

**Comment by escattone** [comment]
@aardrian It should be fully deployed about 5-6 hours from now, and then it's a matter of when each page's CDN cache expires (at most 24 hours). So, at the latest, all pages should show this change 30 hours from now. Cheers.

**Comment by dhananjayd1729** [comment]
> @dhananjayd1729 This is on the critical path for us so I am going to pick this one up. Please have a look at issues marked as good first issues or just let us know if you are interested in picking something else up. I hope you have gotten Yari working locally. Thank you again for your interest in contributing.

@schalkneethling First of all I'm really sorry for being not active on time(there was a lot things going with me lately, my exams and all).But when I said I want to work on this issue that time I tried my best to run yari locally but I couldn't. I would like to contribute to MDN docs for upcoming issues.I hope I will get help when I get stucked.

**Comment by schalkneethling** [comment]
> > @dhananjayd1729 This is on the critical path for us so I am going to pick this one up. Please have a look at issues marked as good first issues or just let us know if you are interested in picking something else up. I hope you have gotten Yari working locally. Thank you again for your interest in contributing.
> 
> @schalkneethling First of all I'm really sorry for being not active on time(there was a lot things going with me lately, my exams and all).But when I said I want to work on this issue that time I tried my best to run yari locally but I couldn't. I would like to contribute to MDN docs for upcoming issues.I hope I will get help when I get stucked.

No need to apologize @dhananjayd1729. Please let us know whenever you are ready and find something you are interested in.

**Comment by dhananjayd1729** [comment]
Sure @schalkneethling . Looking forward to contribute.
